libs/build_hdf5.sh

```diff
@@ -85,3 +87,8 @@ VERBOSE=$MAKE_VERBOSE make -j${NTHREADS:-4}
 [[ -z $mpi ]] && modpath=compiler || modpath=mpi
 $MODULES && update_modules $modpath $name $version
 echo $name $version $URL >> ${HPC_STACK_ROOT}/hpc-stack-contents.log
+# save the current modules list to the hpc-hdf5 collection
```

Comment:
I don't see how this fixes the problem with the CISL modules always taking precedence over modules added by users. Can you explain, please?

Reply:
@climbfuji - storing module collections is a Lmod functionality, it uses `module save <lmod_collection>`   and `module restore <lmod_collection>` commands, and the collections of modules are saved in user's home directory `~/.lmod.d/`.  This is a standard way similar to user space configuration using ~/.ssh/ directories, or ~/.matlab/ , ~/.vim, ~/.gitconfig .
An example of such a collection, the file `~/.lmod.d/hpc-mpi-esmf.ch`  on cheyenne in my home directory, is attached, except with the name modified to allowed format to attache, with the added *.TXT extension.

Additional functionality is used to prevent the system to load default module versions, when a specific version of the same software module is needed and it is already loaded. The syntax uses 'module is-loaded' construct. A couple of  examples:
`module is-loaded hdf5 || module load hdf5`
`module is-loaded cmake || module try-load cmake`
`module is-loaded netcdf || module load netcdf`


[hpc-mpi-esmf.ch.TXT](https://github.com/NOAA-EMC/hpc-stack/files/8637652/hpc-mpi-esmf.ch.TXT)

Reply:
I'm cleaning up the code a little more, to separate the builds with higher/progressive module versions on cheyenne and the added functionality with the module collections, and will update the PR.